appveyor.yml

```
@@ -17,6 +17,9 @@ test_script:
 - git clean -dxf tests
 - msbuild /restore tests/RefVersion3.1.0 /warnaserror /nr:false /v:m
 - git clean -dxf tests
+# No /warnaserror because WarningsNotAsErrors is not respected
+- msbuild /restore tests/NoDisabledWarnings /nr:false /v:m
```

We still want the compiler to escalate warnings produced within CSC:

```suggestion
- msbuild /restore tests/NoDisabledWarnings /p:TreatWarningsAsErrors=true /nr:false /v:m
```

---

@sharwell This would cause the build to fail and would require net35, netcoreapp1.0 and  netcoreapp1.1. Desktop MSBuild is not respecting WarningsNotAsErrors.

---

📝 This is not the same flag as `/warnaserror`. While `/warnaserror` affects everything reported through MSBuild, `TreatWarningsAsErrors` only applies to warnings produced within the C# compiler, and that part should be respecting `WarningsNotAsErrors`.

---

I could be wrong, but we should at least check 😆

---

Won't we still have to remove netcoreapp1.0 and netcoreapp1.1 due to the CS1701 warning if we use TreatWarningsAsErrors?

---

Desktop csc follows WarningsNotAsErrors.

---

Sorry, got it!